controllers/core/event_controller.go

```diff
@@ -0,0 +1,207 @@
+	)
+
+	eventControllerSGPEventsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "security_group_pod_events_reconciled_by_event_controller",
+			Help: "The number of SGP events that were reconcile by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerCNEventsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "custom_networking_events_reconciled_by_event_controller",
+			Help: "The number of custom networking events that were reconcile by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerEventFailureCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "failed_get_events__by_event_controller",
+			Help: "The number of failures on getting events by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventReconcileOperationLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "event_reconcile_operation_latency",
+			Help: "Event controller reconcile events latency in ms",
+		},
+		[]string{"time"},
+	)
+
+	prometheusRegistered = false
+	reconcileEvents      = "reconcile_events"
+)
+
+func PrometheusRegister() {
+	if !prometheusRegistered {
+		metrics.Registry.MustRegister(eventControllerTotalEventsCount)
+		metrics.Registry.MustRegister(eventControllerSGPEventsCount)
+		metrics.Registry.MustRegister(eventControllerEventFailureCount)
+		metrics.Registry.MustRegister(eventControllerCNEventsCount)
+		metrics.Registry.MustRegister(eventReconcileOperationLatency)
+		prometheusRegistered = true
+	}
+}
+
+func (r *EventReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	start := time.Now()
+	var err error
+
+	logger := r.Log.WithValues("event", req.Name)
+
+	eventControllerTotalEventsCount.WithLabelValues("reconcile").Inc()
+
+	ops := []client.ListOption{
+		client.MatchingFields{
+			EventFilterKey: config.VpcCNIReportingAgent,
+		},
+	}
+
+	events, err := r.K8sAPI.ListEvents(ops)
+	if err != nil {
+		logger.Error(err, "List events failed")
+		eventControllerEventFailureCount.WithLabelValues("failing_list").Inc()
+		return ctrl.Result{}, err
+	}
+
+	for _, event := range events.Items {
+		// only check and process events were created in the past EventCreatedPastMinutes minutes
+		if !event.GetCreationTimestamp().After(time.Now().Add(-time.Minute * EventCreatedPastMinutes)) {
+			continue
+		}
+		if r.isValidEventForSGP(event) || r.isValidEventForCustomNetworking(event) {
+			nodeName := string(event.InvolvedObject.UID)
+
+			if node, err := r.K8sAPI.GetNode(nodeName); err != nil {
+				// if not found the node, we don't requeue and just wait VPC CNI to send another request
+				r.Log.V(1).Info("Event reconciler didn't find the node and wait for next request for the node", "Node", node.Name)
+				return ctrl.Result{}, nil
+			} else {
+				if r.isEventToManageNode(event) && r.isValidEventForSGP(event) {
+					r.K8sAPI.AddLabelToManageNode(node, config.HasTrunkAttachedLabel, "true")
```

Comment:
Need to handle error scenario here?

Reply:
Added error handling and unit test for error. Thanks.